dbapi/src/main/java/com/griddynamics/jagger/dbapi/provider/SessionInfoProviderImpl.java

```diff
@@ -273,53 +283,61 @@ public Long getFirstPosition(Set<String> selectedIds) throws RuntimeException {
         checkArgument(length >= 0, "length is negative");
         checkNotNull(sessionIds, "sessionIds is null");
 
-        long timestamp = System.currentTimeMillis();
+        boolean isFetchAll = sessionIds.size() == 0; // if input set is empty - fetch all session data.
 
+        long timestamp = System.currentTimeMillis();
         List<SessionDataDto> sessionDataDtoList;
-
         try {
+            Query query = entityManager.createQuery(
+                    "select sd from SessionData as sd " +
+                    (isFetchAll ? " " : "where sd.sessionId in (:sessionIds) ") +
+                    "order by sd.startTime asc"
+            ).setFirstResult(start);
+            if (!isFetchAll) {
+                query.setParameter("sessionIds", new ArrayList<String>(sessionIds))
+                     .setMaxResults(length);
+            }
             @SuppressWarnings("unchecked")
-            List<SessionData> sessionDataList = (List<SessionData>)
-                    entityManager.createQuery("select sd from SessionData as sd where sd.sessionId in (:sessionIds) order by sd.startTime asc")
-                            .setParameter("sessionIds", new ArrayList<String>(sessionIds))
-                            .setFirstResult(start)
-                            .setMaxResults(length)
-                            .getResultList();
-
+            List<SessionData> sessionDataList = (List<SessionData>) query.getResultList();
             if (sessionDataList.isEmpty()) {
                 return Collections.<SessionDataDto>emptyList();
             }
 
             Map<Long, String> userCommentMap = Collections.EMPTY_MAP;
-
             if (isUserCommentStorageAvailable) {
-
-                List<Object[]> userComments = entityManager.createQuery(
-                        "select smd.sessionData.id, smd.userComment from SessionMetaDataEntity as smd where smd.sessionData in (:sessionDataList)")
-                        .setParameter("sessionDataList", sessionDataList)
-                        .getResultList();
-
+                query = entityManager.createQuery(
+                        "select smd.sessionData.id, smd.userComment from SessionMetaDataEntity as smd "
+                        + (isFetchAll ? " " : "where smd.sessionData in (:sessionDataList)")
+                );
+                if (!isFetchAll) {
+                    query.setParameter("sessionDataList", sessionDataList);
+                }
+                List<Object[]> userComments = query.getResultList();
                 if (!userComments.isEmpty()) {
```

Comment:
I'd better use CollectionUtils.isNotEmpty(userComments), since it's more obvious and clear